openapi/openapi_v2_spec_analyser.go

```diff
@@ -60,20 +73,12 @@ func (specAnalyser *specV2Analyser) GetTerraformCompliantResources() ([]SpecReso
 		}
 		if isMultiRegion {
 			log.Printf("[INFO] resource '%s' is configured with host override AND multi region; creating one reasource per region", resourceRootPath)
-			for regionName := range regions {
-				r, err := newSpecV2ResourceWithRegion(regionName, resourceRootPath, *resourcePayloadSchemaDef, *resourceRoot, pathItem, specAnalyser.d.Spec().Definitions, specAnalyser.d.Spec().Paths.Paths)
-				if err != nil {
-					log.Printf("[WARN] ignoring resource '%s' due to an error while creating a creating the SpecV2Resource: %s", resourceRootPath, err)
-					continue
-				}
-				regionHost, err := r.getHost()
-				if err != nil {
-					log.Printf("multi region host for resource '%s' is not valid: ", err)
-					continue
-				}
-				log.Printf("[INFO] multi region resource name = %s, region = '%s', host = '%s'", r.getResourceName(), regionName, regionHost)
-				resources = append(resources, r)
+			multiRegionResources, err := specAnalyser.createMultiRegionResources(regions, resourceRootPath, *resourceRoot, pathItem, resourcePayloadSchemaDef)
+			if err != nil {
+				log.Printf("[WARN] ignoring multiregion resource '%s' due to an error: %s", resourceRootPath, err)
+				continue
```

Comment:
I would expect that if there's an error from the createMultiRegionResources calls, then multiRegionResources will be an empty slice, and then can safely go ahead with the following `append` call, meaning that we can drop the `continue`.  At that point, if it's not important enough to guard against a regression in the log call, perhaps we can just eliminate the entire `err != nil` conditional branch.